fenix/app/src/main/java/org/mozilla/fenix/compose/HorizontalPagerIndicator.kt

```diff
@@ -0,0 +1,136 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.offset
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.ContentAlpha
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.LocalContentColor
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Shape
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.pager.HorizontalPager
+import com.google.accompanist.pager.PagerState
+import kotlin.math.absoluteValue
+import kotlin.math.sign
+
+/**
+ * An horizontally laid out indicator for a [HorizontalPager], that's almost the same as the
+ * accompanist HorizontalPagerIndicator, but with the ability to leave the trail of indicators
+ * to show progress, instead of just showing the current one as active.
+ *
+ * This element allows the setting of the [indicatorShape], which defines how the
+ * indicator is visually represented.
+ *
+ * @param pagerState the state object of your [HorizontalPager] to be used to observe the list's state.
+ * @param modifier the modifier to apply to this layout.
+ * @param pageCount the size of indicators should be displayed, defaults to [PagerState.pageCount].
+ * If you are implementing a looping pager with a much larger [PagerState.pageCount]
+ * than indicators should displayed, e.g. [Int.MAX_VALUE], specify you real size in this param.
+ * @param pageIndexMapping describe how to get the position of active indicator by the giving page
+ * from [PagerState.currentPage], if [pageCount] is not equals to [PagerState.pageCount].
+ * @param activeColor the color of the active Page indicator
+ * @param inactiveColor the color of page indicators that are inactive. This defaults to
+ * [activeColor] with the alpha component set to the [ContentAlpha.disabled].
+ * @param indicatorWidth the width of each indicator in [Dp].
+ * @param indicatorHeight the height of each indicator in [Dp]. Defaults to [indicatorWidth].
```

Comment:
small nit: `This defaults to [indicatorWidth]."

Reply:
Yes, it's correct, it makes it a square :)